src/Components/ContextMenu.re

```diff
@@ -0,0 +1,257 @@
+    //   fontFamily("seti.ttf"),
+    //   fontSize(Constants.fontSize),
+    //   marginRight(10),
+    //   color(fgColor),
+    // ];
+
+    let label = (~font: UiFont.t, ~theme: Theme.t, ~isFocused) => [
+      fontFamily(font.fontFile),
+      textOverflow(`Ellipsis),
+      fontSize(Constants.fontSize),
+      color(theme.menuForeground),
+      backgroundColor(bg(~theme, ~isFocused)),
+    ];
+  };
+
+  let make:
+    'data.
+    (
+      ~item: item('data),
+      ~theme: Theme.t,
+      ~font: UiFont.t,
+      ~onClick: unit => unit,
+      unit
+    ) =>
+    React.element(React.node)
+   =
+    (~item, ~theme, ~font, ~onClick, ()) => {
+      let isFocused = false;
+
+      // let iconView =
+      //   switch (item.icon) {
+      //   | Some(icon) =>
+      //     IconTheme.IconDefinition.(
+      //       <Text
+      //         style={Styles.icon(icon.fontColor)}
+      //         text={FontIcon.codeToIcon(icon.fontCharacter)}
+      //       />
+      //     )
+
+      //   | None => <Text style={Styles.icon(Colors.transparentWhite)} text="" />
+      //   };
+
+      let labelView = {
+        let style = Styles.label(~font, ~theme, ~isFocused);
+        <Text style text={item.label} />;
+      };
+
+      <Clickable onClick>
+        <View style={Styles.container(~theme, ~isFocused)}>
+          // iconView
+           labelView </View>
+      </Clickable>;
+      // onMouseOut={_ => setIsFocused(_ => false)}
+      // onMouseOver={_ => setIsFocused(_ => true)}
+    };
+};
+
+// MENU
+
+module Menu = {
+  module Styles = {
+    open Style;
+
+    let container = (~x, ~y, ~theme: Theme.t) => [
+      position(`Absolute),
+      top(y),
+      left(x),
+      backgroundColor(theme.menuBackground),
+      color(theme.menuForeground),
+      width(Constants.menuWidth),
+    ];
+  };
+
+  let component = React.Expert.component("Menu");
+  let make = (~model, ~theme, ~font, ~onItemSelect, ()) =>
+    component(hooks => {
+      let ((maybeRef, setRef), hooks) = Hooks.state(None, hooks);
+      let {x, y, originX, originY, items, _} = model;
+
+      let height =
+        switch (maybeRef) {
+        | Some((node: node)) => node#measurements().height
+        | None => List.length(model.items) * 20
+        };
+      let width = Constants.menuWidth;
+
+      let x =
+        switch (originX) {
+        | `Left => x
+        | `Middle => x - width / 2
+        | `Right => x - width
+        };
+
+      let y =
+        switch (originY) {
+        | `Top => y - height
+        | `Middle => y - height / 2
+        | `Bottom => y
+        };
+
+      (
+        // TODO: BoxShadow apparently blocks mouse events. Figure out why before adding back
+        // <BoxShadow
+        //   boxShadow={Style.BoxShadow.make(
+        //     ~xOffset=-11.,
+        //     ~yOffset=-11.,
+        //     ~blurRadius=25.,
+        //     ~spreadRadius=0.,
+        //     ~color=Color.rgba(0., 0., 0., 0.2),
+        //     (),
+        //   )}>
```

Comment:
This is unfortunate. No idea what's going on here, but if this is uncommented the menu items stop receiving mouse events.

Reply:
Bummer, I'll look at bringing the 'style' property back so we can use it w/o interfering with click events

Reply:
Working on bringing it back here: https://github.com/revery-ui/revery/pull/719

This would let us add a `boxShadow` style to the root `<View />`, side-stepping this issue.